src/NextGen/Framework/PaymentGateways/Traits/HandleHttpResponses.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace Give\NextGen\Framework\PaymentGateways\Traits;
+
+use Give\Framework\Http\Response\Types\JsonResponse;
+use Give\Framework\Http\Response\Types\RedirectResponse;
+
+/**
+ * This is used to overwrite the same method for handling responses in GiveWP core within our Next Gen feature plugin.
+ *
+ * Eventually this will be removed and the core method will be used.
+ */
+trait HandleHttpResponses
+{
+    /**
+     * Handle Response
+     *
+     * @unreleased add support for json content-type
+     *
+     * @since 2.18.0
+     *
+     * @param  RedirectResponse|JsonResponse  $type
+     */
+    public function handleResponse($type)
+    {
+        if ($type instanceof RedirectResponse) {
+            if (isset($_SERVER['CONTENT_TYPE']) && str_contains($_SERVER['CONTENT_TYPE'], "application/json")) {
+                wp_send_json([
+                    'data' => [
+                        'redirectUrl' => $type->getTargetUrl()
+                    ]
+                ]);
```

Comment:
@jonwaldstein What do you think about adding response "types" for when we send JSON back to the browser? This way we can check the type and assume the shape of the corresponding data. Types might be:
- `redirect`
- `validation_error`
- `success`

Just a thought. 😄

Reply:
I like that idea

Reply:
I added `redirect` and `success` types.  I'll circle back to validation when the time comes b3e69892